lib/modules/datasource/rubygems/get-rubygems-org.ts

```diff
@@ -64,7 +64,7 @@ export class RubyGemsOrgDatasource extends Datasource {
         contentLength = 0;
         packageReleases = Object.create(null); // Because we might need a "constructor" key
         throw new ExternalHostError(
-          new Error('Rubygems fetch error - need to reset cache')
+          new Error('Rubygems fetch error - '.concat(err.message))
```

Comment:
i think we should just debug log the original error here.

Reply:
The original message of `need to reset cache` is not correct and totally misleading. For a user the error shouldn't be hidden in debug logging, IMHO.

Reply:
there isn't any explanation why this is required? what's your issue to solve?

Reply:
This PR was created after this discussion:
https://github.com/renovatebot/renovate/discussions/14901

Reply:
Why not just `log.debug({err}, 'Error fetching Rubygems');`?

Reply:
* My thought is, that the error should be obvious for the user. So use of log.debug might only display the real error in case debug logging is enabled.
* Current message `Rubygems fetch error - need to reset cache` is not applicable to all errors catched in the block.
* Other datasources (pypi, npm) also forward the original error instead of rewriting the error message raised within the try block. So aligning with npm might be a better option. https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/npm/get.ts#L195

Reply:
I've updated to logger.debug, although not my personal preference :-)